pkg/controller/pcidevice/pcidevice_controller_test.go

```diff
@@ -0,0 +1,41 @@
+package pcidevice
+
+import (
+	"context"
+	"github.com/harvester/pcidevices/pkg/generated/clientset/versioned/fake"
+	"github.com/harvester/pcidevices/pkg/util/fakeclients"
+	"github.com/jaypipes/ghw"
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+const (
+	defaultPCIDeviceSnapshot = "../../../tests/snapshots/linux-amd64-e147d239df014921c6cbb49fbc3d6c41.tar.gz"
+)
+
+func Test_reconcilePCIDevices(t *testing.T) {
+	assert := require.New(t)
+	client := fake.NewSimpleClientset()
+
+	pci, err := ghw.PCI(ghw.WithSnapshot(ghw.SnapshotOptions{
```

Comment:
Our [existing ghw library](https://github.com/jaypipes/ghw) definitely does this too: https://github.com/jaypipes/ghw/blob/main/pkg/option/option.go#L173-L177

Reply:
we are using the same, just have a pre-generated snapshot